src/demikernel/libos/network/libos.rs

```diff
@@ -300,16 +303,23 @@ impl<T: NetworkTransport> SharedNetworkLibOS<T> {
     /// coroutine that asynchronously runs the push and any synchronous multi-queue functionality before the push
     /// begins.
     pub fn push(&mut self, qd: QDesc, sga: &demi_sgarray_t) -> Result<QToken, Fail> {
-        let buf: DemiBuffer = clone_sgarray(sga)?;
-        if buf.is_empty() {
-            let cause: String = String::from("zero-length buffer");
+        let bufs: ArrayVec<DemiBuffer, DEMI_SGARRAY_MAXLEN> = clone_sgarray(sga)?;
+        if bufs.is_empty() {
+            let cause: String = String::from("zero-length list of buffers");
```

Comment:
`String::from` can be replacled with string literals and references to cause can be removed while passing into other blocks.